libpod/container_log.go

```diff
@@ -21,10 +21,31 @@ func init() {
 	logDrivers = append(logDrivers, define.KubernetesLogging, define.NoLogging, define.PassthroughLogging)
 }
 
+func cloneLogOpts(options *logs.LogOptions, newColor int64) *logs.LogOptions {
+	clone := logs.LogOptions{
+		Details:    options.Details,
+		Follow:     options.Follow,
+		Since:      options.Since,
+		Until:      options.Until,
+		Tail:       options.Tail,
+		Timestamps: options.Timestamps,
+		Colors:     options.Colors,
+		ColorID:    newColor,
+		Multi:      options.Multi,
+		WaitGroup:  options.WaitGroup,
+		UseName:    options.UseName,
+	}
+	return &clone
+}
+
 // Log is a runtime function that can read one or more container logs.
 func (r *Runtime) Log(ctx context.Context, containers []*Container, options *logs.LogOptions, logChannel chan *logs.LogLine) error {
-	for _, ctr := range containers {
-		if err := ctr.ReadLog(ctx, options, logChannel); err != nil {
+	for c, ctr := range containers {
+		clonedOpts := options
+		if options.Colors {
+			clonedOpts = cloneLogOpts(options, int64(c))
```

Comment:
I don't think it is a good idea to copy all options just to change one field. Can you add the color field as new parameter to ctr.ReadLog() so we do not have to copy the full struct.